ibc-clients/ics08-wasm/types/src/consensus_state.rs

```diff
@@ -54,21 +54,19 @@ impl From<ConsensusState> for Any {
 }
 
 impl TryFrom<Any> for ConsensusState {
-    type Error = ClientError;
+    type Error = DecodingError;
 
     fn try_from(any: Any) -> Result<Self, Self::Error> {
-        fn decode_consensus_state(value: &[u8]) -> Result<ConsensusState, ClientError> {
-            let consensus_state =
-                Protobuf::<RawConsensusState>::decode(value).map_err(|e| ClientError::Other {
-                    description: e.to_string(),
-                })?;
+        fn decode_consensus_state(value: &[u8]) -> Result<ConsensusState, DecodingError> {
+            let consensus_state = Protobuf::<RawConsensusState>::decode(value)?;
             Ok(consensus_state)
         }
         match any.type_url.as_str() {
             WASM_CONSENSUS_STATE_TYPE_URL => decode_consensus_state(&any.value),
-            _ => Err(ClientError::Other {
-                description: "type_url does not match".into(),
-            }),
+            _ => Err(DecodingError::MismatchedResourceName {
+                expected: WASM_CONSENSUS_STATE_TYPE_URL.to_string(),
+                actual: any.type_url.to_string(),
+            })?,
```

Comment:
I am seeing this pattern often. Maybe we can refactor it as

```rs
if let WASM_CONSENSUS_STATE_TYPE_URL = any.type_url.as_str() {
  Protobuf::<RawConsensusState>::decode(value)
} else {
  Err(...)
}
```

Reply:
Nice suggestion: [91ac57b](https://github.com/cosmos/ibc-rs/pull/1350/commits/91ac57be7636753280ce76df18f1ad2280bbf94d)